entrypoint.sh

```diff
@@ -2,25 +2,65 @@
 # entrypoint.sh
 set -e
 
-COMMAND=${1:-dump-cron}
+if [[ -z $COMMAND ]];
+then
+   COMMAND=${1:-dump-cron}
+fi
+
 CRON_SCHEDULE=${CRON_SCHEDULE:-0 1 * * *}
 PREFIX=${PREFIX:-dump}
 PGUSER=${PGUSER:-postgres}
-PGDB=${PGDB:-postgres}
-PGHOST=${PGHOST:-db}
+POSTGRES_DB=${POSTGRES_DB:-postgres}
+PGHOST=${PGHOST:-localhost}
 PGPORT=${PGPORT:-5432}
+PGDUMP=${PGDUMP:-'/dump'}
+POSTGRES_PASSWORD_FILE=${POSTGRES_PASSWORD_FILE:-'/run/secrets/db_password'}
 
+if [[ -n ${PG_LOG} ]];
+then
+   echo "COMMAND: ${COMMAND}"
+   echo "CRON_SCHEDULE: ${CRON_SCHEDULE}"
+   echo "PREFIX: ${PREFIX}"
+   echo "PGUSER: ${PGUSER}"
+   echo "POSTGRES_DB: ${POSTGRES_DB}"
+   echo "PGHOST: ${PGHOST}"
+   echo "PGPORT: ${PGPORT}"
+   echo "PGDUMP: ${PGDUMP}"
+   echo "POSTGRES_PASSWORD_FILE: ${POSTGRES_PASSWORD_FILE}"
+fi
+
+if [[ -f /run/secrets/db_password ]];
+then
+   source /run/secrets/db_password
+else
+   echo "ERROR: No password file found!"
+   echo "It is suggested that a docker secrets file is used for security concerns."
+   echo "If not, ensure that POSTGRES_PASSWORD is set."
+fi
 
 if [[ "${COMMAND}" == 'dump' ]]; then
+    /usr/local/bin/docker-entrypoint.sh postgres &
+    SLEEPTIME=5
+    echo "Sleeping ${SLEEPTIME} seconds to allow database to stand up..."
+    sleep ${SLEEPTIME}
     exec /dump.sh
 elif [[ "${COMMAND}" == 'dump-cron' ]]; then
-    LOGFIFO='/var/log/cron.fifo'
+
+    if [[ -z ${LOGFIFO} ]];
+    then
+      LOGFIFO=${PGDUMP}/cron.fifo
+    fi
+
+    if [[ -n ${PG_LOG} ]];
+    then
+        echo "LOGFIFO: ${LOGFIFO}"
+    fi
     if [[ ! -e "${LOGFIFO}" ]]; then
         mkfifo "${LOGFIFO}"
     fi
-    CRON_ENV="PREFIX='$PREFIX'\nPGUSER='${PGUSER}'\nPGDB='${PGDB}'\nPGHOST='${PGHOST}'\nPGPORT='${PGPORT}'"
-    if [[ -n "${PGPASSWORD}" ]]; then
-        CRON_ENV="$CRON_ENV\nPGPASSWORD='${PGPASSWORD}'"
+    CRON_ENV="PREFIX='$PREFIX'\nPGUSER='${PGUSER}'\nPOSTGRES_DB='${POSTGRES_DB}'\nPGHOST='${PGHOST}'\nPGPORT='${PGPORT}'\nPGDUMP='${PGDUMP}'"
+    if [[ -n "${POSTGRES_PASSWORD}" ]]; then
```

Comment:
POSTGRES_PASSWORD should be an alias for PGPASSWORD, and listed in the documentation table